### DISCUSSION TITLE
Mapping to the wrong class

### DISCUSSION THREAD

**Discussion by GMQS** [CuyZ/Valinor#392] [Q&A]
I am using it to generate class instances from JSON. However, I have encountered an issue. It involves a simple structured class with an array type property, and I'm using the `array<T>` syntax from phpstan in the documentation comments. In this case, if I don't specify the class to be used for generics with its fully qualified name, it generates instances of the wrong class. From what I can infer, it seems that the behavior doesn't take into account the namespace of the class name specified in the documentation comments. It's likely that my custom `Error` class was being mapped to PHP's standard `Error` class, leading to this issue. For more details, please refer to the sample code below.

## Response Class
```php
namespace MyApp\Types;

use MyApp\Types\Error; // don't recognize ?

final class Response {

     /**
     * @param list<Error> $errors // Mismapping to standard Error class occurs. list<\MyApp\Types\Error> is work.
     */
    public function __construct(public readonly array $errors) {}
}
```

## Error Class
```php
namespace MyApp\Types\Error;

final class Error {
    public function __construct(public readonly string $message) {}
}

```

**Comment by romm** [comment]
Hi @GMQS, thanks for the report! This should be fixed with #398. 😉
